lib/module-types.js

```diff
@@ -0,0 +1,51 @@
+module.exports = (configuredExtensions, defaultModuleType) => {
+	const lockedModuleTypes = {
+		cjs: 'commonjs',
+		mjs: 'module',
+		js: defaultModuleType
+	};
+
+	const createModuleTypesFromExtensionsObject = extensionsObject => {
+		const configuredModuleTypes = {};
+
+		for (const [extension, type] of Object.entries(extensionsObject)) {
+			if (extension in lockedModuleTypes && type !== lockedModuleTypes[extension]) {
```

Comment:
Possibly irrelevant (see below), but I prefer `Reflect.has()` these days. I find `in` very easy to overlook.

```suggestion
			if (Reflect.has(lockedModuleTypes, extension) && type !== lockedModuleTypes[extension]) {
```